--- cmake/modules/FindWPD.cmake.orig	2010-12-28 11:53:19.271867868 +0100
+++ cmake/modules/FindWPD.cmake	2010-12-28 11:57:22.209342685 +0100
@@ -20,11 +20,13 @@
 if(NOT WIN32)
   INCLUDE(FindPkgConfig)
   pkg_check_modules(WPD libwpd-0.8)
+  pkg_check_modules(WPD libwpd-0.9)
 endif(NOT WIN32)
 
   FIND_PATH(WPD_INCLUDE_DIR libwpd/libwpd.h
     ${WPD_INCLUDE_DIR}
     /usr/include/libwpd-0.8
+    /usr/include/libwpd-0.9
   )
 
 
--- cmake/modules/FindWPG.cmake.orig	2010-12-28 11:53:29.508533234 +0100
+++ cmake/modules/FindWPG.cmake	2010-12-28 11:58:43.802668625 +0100
@@ -19,9 +19,10 @@
   IF (NOT WIN32)
     INCLUDE(FindPkgConfig)
     pkg_check_modules(LIBWPG libwpg-0.1)
+    pkg_check_modules(LIBWPG libwpg-0.2)
   ENDIF (NOT WIN32)
 
-  FIND_LIBRARY(LIBWPG_STREAM_LIBRARIES NAMES wpg-stream-0.1 libwpg-stream-0.1
+  FIND_LIBRARY(LIBWPG_STREAM_LIBRARIES NAMES libwpg-0.1 libwpg-0.2
     PATHS
     ${LIBWPG_LIBRARIES}
     )
